crates/net/network/src/transactions/mod.rs

```diff
@@ -2275,6 +2279,131 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_partially_tx_response() {
```

Comment:
can we add a description what this test does,
this test is fairly long so in case we change some internals in future would be nice to some docs here